### ISSUE TITLE
${env} isn't processed in C_Cpp.default.includePath

### ISSUE THREAD

**Issue by ijash** [microsoft/vscode-cpptools#3309]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Linux mint 19.01
- VS Code Version: 1.32.3
- C/C++ Extension Version:0.21
- Other extension: arduino  0.2.25

I can't get the  `C_Cpp.default.IncludePath` get to work. please see details below.

**To Reproduce**

1. i have followed the instruction in https://github.com/Microsoft/vscode-cpptools/blob/master/Documentation/LanguageServer/Customizing%20Default%20Settings.md
2. my `settings.json` is 
```json
{    "C_Cpp.default.includePath": [
        "${env.HOME}/.arduino"
        ]
}
```
my `c_cpp_properties.json` is:
```json
{
    "configurations": [
        {
            "name": "Linux",
            "includePath": [
                "/opt/arduino/arduino-1.8.8/tools/**",
                "/opt/arduino/arduino-1.8.8/hardware/arduino/avr/**",
                "${default}"
            ],
            "forcedInclude": [
                "/opt/arduino/arduino-1.8.8/hardware/arduino/avr/cores/arduino/Arduino.h"
            ],
            "intelliSenseMode": "gcc-x64",
            "compilerPath": "/usr/bin/gcc",
            "cStandard": "c11",
            "cppStandard": "c++17"
        }
    ],
    "version": 4
}
```
mostly auto generated except `${default}`

3. i write in `main.ino` which use C++ language as follows:
```c
#include <MFRC522.h>


```
and got a problem `Include file not found in browse.path.`

4. BUT, when i move the `"${env.HOME}/.arduino"` to `c_cpp_properties.json` inside `configurations`->`includePath` array, the problem disappears.

**Expected behavior**

I expect something like global configuration or default configuration in `settings.json` so i don't have to re-write configuration each new project to `c_cpp_properties.json`. and my expectation is the same as issue https://github.com/Microsoft/vscode-cpptools/issues/368#issue-191469357
 
**Screenshots**

this is the current problem:
![image](https://user-images.githubusercontent.com/30673286/54560174-4cbfd900-49f4-11e9-875f-d0637267f147.png)

this is the behavior after i edit the `c_cpp_properties.json`:
![image](https://user-images.githubusercontent.com/30673286/54560843-d3c18100-49f5-11e9-8de7-947d25cac0a7.png)

**Additional context**

-none

**Comment by sean-mcmanus** [comment]
You mention `systemIncludePath` and `includePath` -- which one are you actually using?

The "Include file not found in browse.path." is when the `browse.path` isn't set corectly (i.e. IntelliSense is not being used). It's possible the bug is that we're not correctly copying over the `C_Cpp.default.includePath` setting into the `browse.path`.

**Comment by ijash** [comment]
> You mention `systemIncludePath` and `includePath` -- which one are you actually using?
> 
> The "Include file not found in browse.path." is when the `browse.path` isn't set corectly (i.e. IntelliSense is not being used). It's possible the bug is that we're not correctly copying over the `C_Cpp.default.includePath` setting into the `browse.path`.

my bad, it's `IncludePath`. i've updated the post.

you mention  : 

not correctly copying over the `C_Cpp.default.includePath` setting into the `browse.path`.

and how to do that?

**Comment by sean-mcmanus** [comment]
Our code is supposed to do that copying -- the bug could be that we don't do that correctly...I'll double check.

**Comment by ijash** [comment]
> Our code is supposed to do that copying -- the bug could be that we don't do that correctly...I'll double check.

thank you, if anything I am able to provide for help, please let me know.

**Comment by sean-mcmanus** [comment]
I reproed the bug -- the problem is we aren't not processing the "${env}" when they're in the "default" settings.

**Comment by sean-mcmanus** [comment]
Thanks for reporting this. We have a fix at https://github.com/Microsoft/vscode-cpptools/pull/3310, but I don't know if it will make into 0.22.0 yet (otherwise, it'll be in the release after that).

**Comment by ijash** [comment]
> Thanks for reporting this. We have a fix at #3310, but I don't know if it will make into 0.22.0 yet (otherwise, it'll be in the release after that).

thank you for updating this issue, looking forward in the next linux release.

**Comment by sean-mcmanus** [comment]
0.22.0 has the fix for `${env.HOME}` not working in the "default.*" settings.
